client/engine/chainservice/mockchain.go

```diff
@@ -0,0 +1,71 @@
+package chainservice
+
+import (
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// MockChain provides an interface which simulates a blockchain network. It is designed for use as a central service which multiple
+// ChainServices connect to with Go chans.
+// It keeps a record of of holdings and adjudication status for each channel, accepts transactions and emits events.
+type MockChain struct {
+	out map[types.Address]chan Event // out is a mapping with a chan for each connected ChainService, used to send Events to that service
+	in  chan protocols.Transaction   // in is the chan used to recieve Transactions from multiple ChainServices
+
+	holdings map[types.Destination]types.Funds // holdings tracks funds for each channel
+}
+
+// Out() returns the out chan for a particular ChainService, and narrows the type so that external consumers mays only recieve on it.
```

Comment:
```suggestion
// Out returns the out chan for a particular ChainService, and narrows the type so that external consumers may only receive on it.
```